saleor/plugins/invoicing/plugin.py

```diff
@@ -0,0 +1,35 @@
+from typing import Any, Optional
+from uuid import uuid4
+
+from django.core.files.base import ContentFile
+
+from ...core import JobStatus
+from ...invoice.models import InvoiceJob
+from ...order.models import Order
+from ..base_plugin import BasePlugin
+from .utils import generate_invoice_number, generate_invoice_pdf
+
+
+class InvoicingPlugin(BasePlugin):
+    PLUGIN_ID = "mirumee.invoicing"
+    PLUGIN_NAME = "Invoicing"
+    DEFAULT_ACTIVE = True
+    PLUGIN_DESCRIPTION = "Built-in saleor plugin that handles invoice creation."
+
+    def invoice_request(
+        self,
+        order: "Order",
+        invoice_job: "InvoiceJob",
+        number: Optional[str],
+        previous_value: Any,
+    ) -> Any:
+        file_content, creation_date = generate_invoice_pdf(invoice_job.invoice)
+        invoice_job.invoice.created = creation_date
+        invoice_job.invoice.update_invoice(number=generate_invoice_number())
+        invoice_job.invoice.save()
```

Comment:
Save should be used with `update_fields`. 

(`update_invoice` this function use `save`. That's the reason why we should avoid it)